src/containers/HomeContainer/Home.js

```diff
@@ -128,6 +131,52 @@ const Home = ({ starred, navigation }) => {
     )
   }
 
+  /**
+   * Render department's course offerings in a dropdown picker based on department id.
+   */
+  const renderDepartmentsDropDown = () => {
+    const [departments, setAllDepartments] = useState([])
+    useEffect(() => {
+      const fetchDepartments = async () => {
+        const allDept = await getUniversityDepartments(universityId)
+        setAllDepartments(allDept)
+      }
+      fetchDepartments()
+    }, [setAllDepartments])
+
+    const departmentItems = departments.map((dept) => {
+      return <Picker.Item key={dept.id} value={dept.id} label={dept.name} />
+    })
+
+    const [department, setDepartment] = useState(departmentId)
+    const onDepartmentSelected = async (newDepartmentId) => {
+      setDepartment(newDepartmentId)
+      departmentId = newDepartmentId
+
+      const allCourses = await getOfferingsData()
+      const newCourses = filterCourses(allCourses)
+
+      try {
+        setCourses(newCourses)
```

Comment:
Why is this in a try catch? Shouldn't this never throw an error?

Reply:
I think one of the tests was throwing a complaint when we didn't have it in a try catch, but I the issue was something else if I remember right. Let me try it out without the try catch.